### PR TITLE
Fail if dartfmt fails for some reason

### PR DESCRIPTION
**Pull request by kevmoo**
Closes https://github.com/travis-ci/travis-ci/issues/7632

### CODE REVIEW

**Comment by kevmoo** [comment]
CC @a14n @nex3 @BanzaiMan

File: lib/travis/build/script/dart.rb [hunk 1]
```diff
@@ -236,6 +236,10 @@ def dartfmt
             end
 
             sh.cmd 'unformatted=`dartfmt -n .`'
+            # If `dartfmt` fails for some reason
+            sh.if '$? -ne 0' do
+              sh.failure ""
+            end
             sh.if '! -z "$unformatted"' do
               sh.echo "Files are unformatted:", ansi: :red
               sh.echo "$unformatted"
```
Comment (kevmoo) [inline lib/travis/build/script/dart.rb]: I'd love to trim the content of `unformatted` too. Any ideas?

File: lib/travis/build/script/dart.rb [hunk 2]
```diff
@@ -236,6 +236,10 @@ def dartfmt
             end
 
             sh.cmd 'unformatted=`dartfmt -n .`'
```
Comment (BanzaiMan) [inline lib/travis/build/script/dart.rb]: `$unformatted` is set at run time by this assignment.

If you want to trim it, you'd have to come up with a `bash` way to trim it (with `tr`, `grep`, `head`, `tail`, `sed`, whatever).

Comment (kevmoo) [inline lib/travis/build/script/dart.rb]: Let's not worry about that for now. I just want to land this so folks stop getting a false positive from `dartfmt`

**Comment by BanzaiMan** [comment]
OK. We'll deploy this Tuesday next week.

**Pull request merged**